mlflow/R/mlflow/R/tracking-experiments.R

```diff
@@ -0,0 +1,209 @@
+#' Create Experiment
+#'
+#' Creates an MLflow experiment.
+#'
+#' @param name The name of the experiment to create.
+#' @param artifact_location Location where all artifacts for this experiment are stored. If
+#'   not provided, the remote server will select an appropriate default.
+#' @template roxlate-client
+#' @export
+mlflow_create_experiment <- function(name, artifact_location = NULL, client = NULL) {
+  client <- client %||% mlflow_client()
+  name <- forge::cast_string(name)
+  response <- mlflow_rest(
+    "experiments", "create",
+    client = client, verb = "POST",
+    data = list(
+      name = name,
+      artifact_location = artifact_location
+    )
+  )
+  mlflow_get_experiment(client = client, experiment_id = response$experiment_id)
+}
+
+#' List Experiments
+#'
+#' Gets a list of all experiments.
+#'
+#' @param view_type Qualifier for type of experiments to be returned. Defaults to `ACTIVE_ONLY`.
+#' @template roxlate-client
+#' @export
+mlflow_list_experiments <- function(view_type = c("ACTIVE_ONLY", "DELETED_ONLY", "ALL"), client = NULL) {
+  client <- client %||% mlflow_client()
+  view_type <- match.arg(view_type)
+  response <-   mlflow_rest(
+    "experiments", "list",
+    client = client, verb = "GET",
+    query = list(view_type = view_type)
+  )
+
+  # Return `NULL` if no experiments
+  if (!length(response)) return(NULL)
+
+  response$experiments %>%
+    purrr::transpose() %>%
+    purrr::map(unlist) %>%
+    tibble::as_tibble()
+}
+
+#' Get Experiment
+#'
+#' Gets metadata for an experiment and a list of runs for the experiment.
+#'
+#'
+#' @param name The experiment name, either this or `experiment_id` should be specified.
+#' @param experiment_id Identifer to get an experiment. Attempts to obtain the active experiment
+#'   if not provided.
+#' @template roxlate-client
+#' @export
+mlflow_get_experiment <- function(name = NULL, experiment_id = NULL, client = NULL) {
+  if (!is.null(name) && !is.null(experiment_id)) {
```

Comment:
should we attempt to infer the experiment if no id nor name is provided?

Reply:
yeah I think it makes sense to return the active experiment or whatever the user set in the env variable?

Reply:
Yeah, sounds good to me.